cmd/logstash/lib/logstash/outputs/loki/entry.rb

```diff
@@ -1,13 +1,24 @@
-module LogStash
-    module Outputs
-        class Loki 
-            class Entry
-                attr_reader :labels, :entry
-                def initialize(labels, entry)
-                    @labels   = labels
-                    @entry = entry
-                end 
-            end 
-        end 
-    end 
-end 
\ No newline at end of file
+module Loki
+    def to_ns(s)
+        (s.to_f * (10**9)).to_i
+    end
+    class Entry
+        attr_reader :labels, :entry
+        def initialize(event,message_field)
+            @entry = {
+                "ts" => to_ns(event.get("@timestamp")),
+                "line" => event.get(message_field).to_s
+            }
+            event = event.clone()
+            event.remove(message_field)
+            event.remove("@timestamp")
+
+            @labels = {}
+            event.to_hash.each { |key,value|
+                next if key.start_with?('@')
+                next if value.is_a?(Hash)
```

Comment:
We should consider the nested fields as well right? Any reason to skip it?

Reply:
They can already create new field out of current field if they want to.

I think logstash is good enough for managing fields that we shouldn't try to do anything.